src/components/custom-field-input-text/custom-field-input-text.test.jsx

```diff
@@ -99,6 +130,25 @@ describe('CustomFieldInputText', () => {
     });
   });
 
+  describe('step API', () => {
+    it('sets the step attribute', () => {
+      render(<TestComponent step={2} />);
+      expect(screen.getByLabelText('Test label')).toHaveAttribute('step', '2');
+    });
+  });
+
+  describe('type API', () => {
+    it('can be set to `number`', () => {
+      render(<TestComponent type="number" />);
+      expect(screen.getByLabelText('Test label')).toHaveAttribute('type', 'number');
+    });
+
+    it('can be set to `text`', () => {
+      render(<TestComponent type="text" />);
+      expect(screen.getByLabelText('Test label')).toHaveAttribute('type', 'text');
+    });
+  });
```

Comment:
Seems like we should have a fail case tested for here as well.

Reply:
I'll repeat what I said in the other similar feedback for the sake of posterity: I don't think we should be testing invalid prop-types. Otherwise, we will be adding a lot of tests for the sake of making sure prop-types are working. Though, I do like that we are testing every (reasonable) valid prop-type since we care about ensuring those behaviors are working.